File: persistent-storage/src/main/java/com/nikita/popularmovies/common/models/MovieDetails.java
package com.nikita.popularmovies.common.models;

import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

import java.util.Collections;
import java.util.List;

@Entity
public final class MovieDetails {
  @PrimaryKey
  private final String id;
  @Embedded
  private final MoviePreview moviePreview;
  private final List<Video> videos;
  private final List<Review> reviews;

  public MovieDetails(MoviePreview moviePreview) {
    this.id = moviePreview.id;
    this.moviePreview = moviePreview;
    videos = Collections.emptyList();
    reviews = Collections.emptyList();
  }

  public MovieDetails(MoviePreview moviePreview, List<Video> videos, List<Review> reviews) {
    this.id = moviePreview.id;
    this.moviePreview = moviePreview;
    this.videos = videos;
    this.reviews = reviews;
  }

  public MoviePreview getMoviePreview() {
    return moviePreview;
  }

  public List<Video> getVideos() {
    return videos;
  }

  public List<Review> getReviews() {
    return reviews;
  }
}

File: persistent-storage/src/main/kotlin/com/nikita/popularmovies/common/models/MovieDetails.kt
package com.nikita.popularmovies.common.models

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.ForeignKey
import android.arch.persistence.room.PrimaryKey

@Entity(foreignKeys = arrayOf(ForeignKey(entity = MovieDetails::class, parentColumns = arrayOf("id"), childColumns = arrayOf("movie_id"))))
data class Review(@field:PrimaryKey val id: String,
                  @field:ColumnInfo(name = "movie_id") val movieId: String,
                  val author: String,
                  val content: String,
                  val url: String)

@Entity(foreignKeys = arrayOf(ForeignKey(entity = MovieDetails::class, parentColumns = arrayOf("id"), childColumns = arrayOf("movie_id"))))
data class Video(@field:PrimaryKey val id: String,
                 @field:ColumnInfo(name = "movie_id") val movieId: String,
                 val key: String,
                 val name: String,
                 val site: String,
                 val type: String)